README.md

```diff
@@ -37,9 +65,63 @@ To run benchmarks, call:
 
 ## Latest benchmarks
 
-    BenchmarkEncodeLevel2-4   3000000  477 ns/op  120 B/op  6 allocs/op
-    BenchmarkEncodeLevel6-4   3000000  515 ns/op  128 B/op  6 allocs/op
-    BenchmarkEncodeLevel15-4  2000000  595 ns/op  176 B/op  6 allocs/op
-    BenchmarkDecodeLevel2-4   5000000  374 ns/op   72 B/op  4 allocs/op
-    BenchmarkDecodeLevel6-4   3000000  403 ns/op   80 B/op  4 allocs/op
-    BenchmarkDecodeLevel15-4  3000000  455 ns/op   99 B/op  4 allocs/op
+    BenchmarkEncode-4        10000000   191 ns/op   0 B/op   0 allocs/op
+    BenchmarkDecode-4         3000000   478 ns/op   3 B/op   1 allocs/op
+    BenchmarkPosition_Code-4  5000000   328 ns/op   32 B/op  1 allocs/op
+
+## Encoding details
+
+A quick explanation of how lat/lon coordinates are encoded to hexagon positions:
+
+Lat/Lon coordinates are projected to a [0,1)x[0x1) square map using
+[Mercator projection](https://en.wikipedia.org/wiki/Mercator_projection), note
+that this means that hexagons at different latitudes cover different Earth areas.
+The coordinates on this square are called `e`, `n` in the code (east, north).
+
+Then we transform those coordinates into `x`, `y` coordinates on the hexagons map:
+we turn the axes 45 degrees and we stretch one of them by a factor of tan(pi/6),
+this makes each four adjacent hexagons centers' be equidistant. Then we decide
+which hexagon contains the desired coordinates. This is probably the trickiest part,
+we use this condition to decide:
+```go
```

Comment:
minor: I think code blocks needs spaces above and below to render in markdown correctly